cupy/cuda/compiler.py

```diff
@@ -531,36 +629,27 @@ def _compile_with_cache_cuda(
         base = _preprocess('', options, arch, backend)
         _empty_file_preprocess_cache[env] = base
 
-    key_src = '%s %s %s %s' % (env, base, source, extra_source)
-    key_src = key_src.encode('utf-8')
-    name = _hash_hexdigest(key_src) + '.cubin'
-
-    mod = function.Module()
+    key = (env, base, source, extra_source, _pickle_protocol)
+    key_str = ('{} '*len(key)).format(*key)
```

Comment:
Let's use faster method :)
```suggestion
    key = (env, base, source, extra_source, str(_pickle_protocol))
    key_str = ' '.join(key)
```